Library/Homebrew/cask/cask.rb

```diff
@@ -19,10 +19,48 @@ class Cask
     extend Searchable
     include Metadata
 
-    attr_reader :token, :sourcefile_path, :source, :config, :default_config, :loaded_from_api
+    HOME_PLACEHOLDER = "$HOME"
+    HOMEBREW_PREFIX_PLACEHOLDER = "$HOMEBREW_PREFIX"
+    APPDIR_PLACEHOLDER = "$APPDIR"
+
+    # TODO: can be removed when API JSON is regenerated with HOMEBREW_PREFIX_PLACEHOLDER.
+    HOMEBREW_OLD_PREFIX_PLACEHOLDER = "$(brew --prefix)"
+
+    attr_reader :token, :sourcefile_path, :source, :config, :default_config, :loaded_from_api, :loader
 
     attr_accessor :download, :allow_reassignment
 
+    class << self
+      def generating_hash!
+        return if generating_hash?
+
+        # Apply monkeypatches for API generation
+        @old_homebrew_prefix = HOMEBREW_PREFIX
+        @old_home = Dir.home
+        Object.send(:remove_const, :HOMEBREW_PREFIX)
+        Object.const_set(:HOMEBREW_PREFIX, Pathname(HOMEBREW_PREFIX_PLACEHOLDER))
+        ENV["HOME"] = HOME_PLACEHOLDER
+
+        @generating_hash = true
+      end
+
+      def generated_hash!
+        return unless generating_hash?
+
+        # Revert monkeypatches for API generation
+        Object.send(:remove_const, :HOMEBREW_PREFIX)
+        Object.const_set(:HOMEBREW_PREFIX, @old_homebrew_prefix)
+        ENV["HOME"] = @old_home
+
+        @generating_hash = false
+      end
+
+      def generating_hash?
+        @generating_hash ||= false
+        @generating_hash == true
+      end
+    end
```

Comment:
How will this handle getting called twice? I ask because the `Cask#to_hash_with_versions` calls `Cask#to_h` internally. Maybe just call it in that method after calling `#to_h`?

Reply:
@apainintheneck The `generating_hash?` guards mean it'll be fine being called twice.